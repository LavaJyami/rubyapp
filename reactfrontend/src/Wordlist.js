import React, { Component } from 'react';

class Wordlist extends Component{
  render(){
    return(
      <div id="wordList">
      <ul className = "wordList">
        {this.props.value.map(item => (
          <li key={item}>{item}</li>
        ))}
          </ul>
          </div>
      );

  }
}

export default Wordlist;
