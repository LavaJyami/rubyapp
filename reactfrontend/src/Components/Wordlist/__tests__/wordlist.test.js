import React from 'react';
import ReactDOM from 'react-dom';
import Wordlist from '../Wordlist';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from 'react-test-renderer';

afterEach(cleanup);

var wordlistArr = ["apple", "oranges", "game", "pc"]
it("Word list renders without crashing",() => {
  const div = document.createElement("div");
  ReactDOM.render(<Wordlist value={wordlistArr} />, div);
});

it("Words in the word list component are rendered properly",() => {
  const {getByTestId} = render(<Wordlist value = {wordlistArr}/>);
  expect (getByTestId('wordlist')).toHaveTextContent("apple");
});

it("matches snapshot",  () => {
  const tree = renderer.create(<Wordlist value = {wordlistArr} />).toJSON();
  expect(tree).toMatchSnapshot();
});
