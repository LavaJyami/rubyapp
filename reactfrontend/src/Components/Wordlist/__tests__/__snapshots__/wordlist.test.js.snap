// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="wordlist"
  id="wordList"
>
  <ul
    className="wordList"
  >
    <li>
      apple
    </li>
    <li>
      oranges
    </li>
    <li>
      game
    </li>
    <li>
      pc
    </li>
  </ul>
</div>
`;
